Various Improvements

askvortsov1 opened the pull request:
- Add option to make the blog the homepage
- Add internationalization to admin
- Use beta 15 extenders
- Add setting to add a link to the blog to the sidebar navigation
I've split this out into granular commits to make review easier

pboschma reviewed (approved):
So, I couldn't test this but I'm sure @jaspervriends will. I like your changes a lot. You took in the requests from the users on the discuss topic and even added translation. Thank you for contributing to this extension.

askvortsov1 commented:
Updated with a few more changes:

- Added Index Page's sideNav widget below the categories nav. Adjusted the style slightly to make it more consistent with the categories widget
- Reimplemented hiding blog tags from the sideNav via JS, removed the extender implementation
- Fixed an issue with the tag redirector
- Added automatic linting and building dist JS

Technically all of these could be separate PRs, but I've concentrated them here. I can split it needed. I've kept commits granular, so this should be merged normally, not via squash, to prevent a very big, confusing commit.

jaspervriends commented:
Wow, that's amazing! I'll go through the changes in a few hours as soon I have some time! 

The IndexSideNav idea sounds as an interesting idea, hadn't thought of that yet, I'm curious for the endresult 😄

Also it's great to see you updated the extenders for beta 15 already, I hadn't time for that yet and left it as it wasn't officially deprecated yet, but I'm glad that's fixed now 😎 

No need to split to multiple PRs by the way, this is fine by me :)

askvortsov1 commented:
> as it wasn't officially deprecated yet

All the events that were replaced with extenders are immediately deprecated, actually :stuck_out_tongue:. For that matter, `Compat` has been deprecated since ~beta 8, we just need to finish up the last few extenders (model visibility, search mainly) before that can be removed.

Oh, and I also added night mode support via less variables for colors (the old .dark css class method isn't used anymore, nightmode uses Flarum's native dark mode). I tried to match them up as best I can, but something might have slipped through.

jaspervriends commented on less/Forum/Item.less:
```diff
@@ -167,12 +167,12 @@
       }
```
Could you add a `>` before the `li`, making it look like: `> li`? Because the forum sidebar nav items are currently receiving a `margin-bottom` as well. (line 164)

jaspervriends reviewed (changes requested):
Although I did find some issues, you improved the extension a lot! Thanks for that!
